### PR TITLE
Run3 hcx249 Test and verify HcalParameter correctly extracted in the dd4hep scenario

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Only applicable for dd4hep interface. It makes use of #28863 

#### PR validation:

Tested with cfg files in the test area Geometry/HcalCommonData/test/python

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalSimParametersFromDD.cc [hunk 1]
```diff
@@ -146,10 +146,11 @@ bool HcalSimParametersFromDD::build(const cms::DDCompactView* cpv, HcalSimulatio
   fv2.mergedSpecifics(ref2);
 
   while (fv2.firstChild()) {
-    const std::string name{fv2.name().data(), fv2.name().size()};
     const std::string matName{cms::dd::noNamespace(fv2.materialName()).data(),
                               cms::dd::noNamespace(fv2.materialName()).size()};
-    if (!isItHF(name, php) &&
+    std::vector<int> copy = fv2.copyNos();
+    int idet = (copy.size() > 1) ? (copy[1] / 1000) : 0;
+    if (((idet == 3) || (idet == 4)) &&
```
Comment (cvuosalo) [inline Geometry/HcalCommonData/src/HcalSimParametersFromDD.cc]: Could you please use constants for 3 and 4? Or at least add a comment that explains what they are?

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalSimParametersFromDD.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/src/HcalSimParametersFromDD.cc]: added a comment line

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
hold
@bsunanda @ianna you wrote this PR makes use of #28863, but #28863 has been closed.
Could you comment on the status of this PR and #28863?

**Comment by bsunanda** [comment]
@silviodonato Please do not hold this. the content of the code which are changed here will remain unchanged irrespective of Ianna's PR is merged or not. I may have to modify some xml files later. These changes are only for dd4hep - it is not going to affect any workflows. I may have to change some xml files and then I have to test the effect of the changes on DDD and DD4Hep

**Comment by silviodonato** [comment]
unhold

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
@fabiocos I noticed there is a commit `Merged refs/pull/28863/head from repository cms-sw with cms-merge-topic`. Luckily,  #28863 has not been merged yet, otherwise this commit would have created problems in the CMSSW history. Is it correct?

**Pull request merged**